### PR TITLE
[EPMCDLAB-1718]: Added Swagger tool to project

### PR DESCRIPTION
**Pull request by ochaparin**

### CODE REVIEW

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/BackupResource.java [hunk 1]
```diff
@@ -37,6 +39,8 @@
 @Slf4j
 @Path("/infrastructure/backup")
 @RolesAllowed("/api/infrastructure/backup")
+@Api(value = "Backup service", authorizations = {@Authorization(SwaggerConfigurator.BASIC_AUTH),
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/BackupResource.java]: Remove basic auth annotation as we currently does not support it

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/SecurityResource.java [hunk 2]
```diff
@@ -51,6 +53,8 @@
 @Path("/user")
 @Consumes(MediaType.APPLICATION_JSON)
 @Produces(MediaType.APPLICATION_JSON)
+@Api(value = "Authorization service", authorizations = {@Authorization(SwaggerConfigurator.BASIC_AUTH),
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/SecurityResource.java]: /login method is not secured

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/BackupResource.java (hunk 1 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/BackupResource.java]: Removed

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/SecurityResource.java [hunk 3]
```diff
@@ -76,7 +80,9 @@ public SecurityResource(SecurityDAO dao, @Named(ServiceConsts.SECURITY_SERVICE_N
      */
     @POST
     @Path("/login")
-    public Response userLogin(@Valid @NotNull UserCredentialDTO credential) {
+	@ApiOperation(value = "Login attempt for user")
```
[the remark this answers is not in the document]
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/SecurityResource.java]: Fixed

File: services/self-service/src/main/java/com/epam/dlab/backendapi/swagger/SwaggerConfigurator.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import io.swagger.jaxrs.config.SwaggerContextService;
+import io.swagger.models.Scheme;
+import io.swagger.models.Swagger;
+import io.swagger.models.auth.BasicAuthDefinition;
+import io.swagger.models.auth.In;
+
+import javax.ws.rs.core.MediaType;
+import java.util.Arrays;
+
+public class SwaggerConfigurator extends SwaggerBundle<SelfServiceApplicationConfiguration> {
```
[the remark this answers is not in the document]
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/swagger/SwaggerConfigurator.java]: Removed

File: services/self-service/src/main/java/com/epam/dlab/backendapi/swagger/SwaggerConfigurator.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+	private static SwaggerConfigurator instance;
+	private Swagger swagger;
+	public static final String BASIC_AUTH = "Basic authorization";
+	public static final String TOKEN_AUTH = "Token authorization";
+
+
+	private SwaggerConfigurator() {
+		swagger = new Swagger();
```
[the remark this answers is not in the document]
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/swagger/SwaggerConfigurator.java]: Refactored

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/SecurityResource.java [hunk 6]
```diff
@@ -123,7 +133,9 @@ public Response authorize(@Auth UserInfo userInfo, @Valid @NotBlank(groups = Aws
      */
     @POST
     @Path("/logout")
-    public Response userLogout(@Auth UserInfo userInfo) {
+	@ApiOperation(value = "Logout attempt for user")
+	@ApiResponses(value = @ApiResponse(code = 403, message = "Logout failed"))
```
[the remark this answers is not in the document]
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/SecurityResource.java]: Added 'success' status codes

**Pull request merged**